hanish/exceptions.py

```diff
@@ -0,0 +1,66 @@
+# hanish.exceptions
+# Exceptions hierarchy for the hanish application.
+#
+# Author:   Benjamin Bengfort <benjamin@bengfort.com>
+# Created:  Mon May 08 12:44:41 2017 -0400
+#
+# Copyright (C) 2016 Bengfort.com
+# For license information, see LICENSE.txt
+#
+# ID: exceptions.py [] benjamin@bengfort.com $
+
+"""
+Exceptions hierarchy for the hanish application.
+"""
+
+##########################################################################
+## Exception Hierarchy
+##########################################################################
+
+class HanishException(Exception):
```

Comment:
can you provide a brief PR comment on when its appropriate/useful to create additional subclasses of exceptions versus just using the attributes the class already exposes (eg msg=)

👍 on the general thoroughness of handling these though